release/internal/version/version.go

```diff
@@ -114,7 +114,7 @@ func DetermineReleaseVersion(v Version, devTagSuffix string) (Version, error) {
 	if HasDevTag(v, devTagSuffix) {
 		// This is the first release from this branch - we can simply extract the version from
 		// the dev tag.
-		return New(strings.Split(gitVersion, devTagSuffix)[0]), nil
+		return New(strings.Split(gitVersion, "-"+devTagSuffix)[0]), nil
```

Comment:
So, this PR broke this right here: https://github.com/projectcalico/calico/commit/5bcda09698e2ec73a8ec2ac06d8ca8cfb1a18a9f#diff-9c76a8ea05e8e2f8e1d03d5e04ac386dce3d9ba16e0f410ee965bdd63f5133eaL91-R114

We used to have logic that prepended the prefix if it didn't exist. I think we probably still want that? It makes this code more reliable (in case someone passes a prefix that already has a "-" on it. 

Basically instead of always including the "-", it should have logic like this:

```go
	if !strings.HasPrefix(devTagSuffix, "-") {
		// The dev tag marker should start with a hyphen.
		// For example in "v3.15.0-0.dev-1-g1234567", we want to split on the "-0.dev" part.
		devTagSuffix = "-" + devTagSuffix
	}
```